### PR TITLE
Compound statements with variable number of select statements

### PR DESCRIPTION
**Pull request by trueqbit**
This PR extends the operators for compound statements (i.e. `union_, union_all, intersect, except`)
-    to accept a variable number of single select statements, which is especially useful for `UNION ALL`,
-    to allow the select statements to return a common type.

Additionally:
- Simplified code for the calculation of node tuples.
- Improved unit tests for node tuples.

My motivating use case is to be able to handpick multiple value ranges:

```c++
struct Object {
    int64 id;
    int64 difference;   // diff value in days
};

union_all(
    select(cast<int64>(min(&Object::difference))),
    select(count(), where(c(&Object::difference) < 0)),
    select(count(), where(c(&Object::difference) >= 0 and c(&Object::difference) < 7)),
    select(count(), where(c(&Object::difference) >= 7 and c(&Object::difference) < 15)),
    ...
);
```

### CODE REVIEW

File: dev/tuple_helper/tuple_iteration.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace sqlite_orm {
 
 #if defined(SQLITE_ORM_FOLD_EXPRESSIONS_SUPPORTED) && defined(SQLITE_ORM_IF_CONSTEXPR_SUPPORTED)
         template<bool reversed = false, class Tpl, size_t... Idx, class L>
-        void iterate_tuple(const Tpl& tpl, std::index_sequence<Idx...>, L&& lambda) {
+        void iterate_tuple(Tpl& tpl, std::index_sequence<Idx...>, L&& lambda) {
```
Comment (fnc12) [inline dev/tuple_helper/tuple_iteration.h:38]: why did you remove `const`? if we don't modify the tuple then we don't ned it as mutable

File: tests/prepared_statement_tests/update_all.cpp [hunk 2]
```diff
@@ -39,15 +39,6 @@ TEST_CASE("Prepared update all") {
 
     {  //  by val
         auto statement = storage.prepare(update_all(set(assign(&User::name, conc(&User::name, "_")))));
-        using Statement = decltype(statement);
```
Comment (fnc12) [inline tests/prepared_statement_tests/update_all.cpp:42]: why did you remove these static parts of ut?

File: dev/column_result.h [hunk 3]
```diff
@@ -220,20 +227,18 @@ namespace sqlite_orm {
         struct column_result_t<DBOs, column_pointer<T, F>, void> : column_result_t<DBOs, F> {};
 
         template<class DBOs, class... Args>
-        struct column_result_t<DBOs, columns_t<Args...>, void> {
-            using type = tuple_cat_t<tuplify_t<column_result_of_t<DBOs, std::decay_t<Args>>>...>;
-        };
+        struct column_result_t<DBOs, columns_t<Args...>, void>
+            : conc_tuple<tuplify_t<column_result_of_t<DBOs, std::decay_t<Args>>>...> {};
```
[the remark this answers is not in the document]
Comment (trueqbit) [reply dev/column_result.h:231]: 🙏

File: tests/prepared_statement_tests/update_all.cpp (hunk 2 above)
Comment (trueqbit) [reply tests/prepared_statement_tests/update_all.cpp:42]: They are covered by the tests already in tests/bindables.cpp and the new "set assign" section in [tests/node_tuple.cpp](https://github.com/fnc12/sqlite_orm/pull/1241/files/9813ec16425e3e4f070a7c22ddafb5cfa16d792f?diff=split&w=0#diff-567fe4ab260b49c101b491d47de299ea1da4e62338918924d86e98c91b8d51fb).

File: dev/tuple_helper/tuple_iteration.h (hunk 1 above)
Comment (trueqbit) [reply dev/tuple_helper/tuple_iteration.h:38]: `compound_operator`'s contructor [sets](https://github.com/fnc12/sqlite_orm/blob/9813ec16425e3e4f070a7c22ddafb5cfa16d792f/dev/select_constraints.h#L121) the `highest_level` flag of the single select-statements.

**Review by fnc12 — APPROVED** [review]
